scripts/ingest_GWAS_Catalog.py

```diff
@@ -0,0 +1,288 @@
+    logging.info(f'Assocation: {parsed_associations.show(2, False, True)}')
+
+    return parsed_associations
+
+def map_associations(parsed_associations: DataFrame) -> DataFrame:
+    # Loading variant annotation and join with parsed associations:
+    logging.info('Loading variant annotation:')
+
+    # Reading and joining variant annotation:
+    variants = (
+        spark.read.parquet(VARIANT_ANNOTATION)
+        .select(
+            f.col('chrom_b38').alias('chr_id'),
+            f.col('pos_b38').alias('chr_pos'),
+            f.col('rsid').alias('rsid_gnomad'),
+            f.col('ref').alias('ref'),
+            f.col('alt').alias('alt')
+        )
+    )
+
+    mapped_associations = (
+        parsed_associations
+        .join(variants, on=['chr_id', 'chr_pos'], how='left_outer')
+        .persist()
+    )
+
+    return mapped_associations
+
+def main():
+
+    # Read GWAS associations and process:
+    gwas_associations = read_GWAS_associations()
+    parsed_gwas_associations = parse_associations(gwas_associations)
+
+    # Debug: save data parsed associations:
+    # parsed_gwas_associations.write.mode('overwrite').parquet(f'{OUTPUT_PATH}/parsed_associations.parquet')
+
+    # Mapping GWAS Catalog variants to GnomAD variants:
+    mapped_associations = map_associations(parsed_gwas_associations)
+
+    # Debug: save mapped data:
+    # mapped_associations.write.mode('overwrite').parquet(f'{OUTPUT_PATH}/mapped_parsed_associations.parquet')
+
+    # Debug: read mapped associations:
+    # mapped_associations = spark.read.parquet(f'{OUTPUT_PATH}/mapped_parsed_associations.parquet').persist()
+
+    # Drop association/variant mappings, where the alleles are discordant:
+    mapped_associations = (
+        mapped_associations
+
+        # Dropping associations with no mapped variants:
+        .filter(f.col('alt').isNotNull())
+
+        # Adding column with the reverse-complement of the risk allele:
+        .withColumn(
+            'risk_allele_reverse_complement',
+            f.when(
+                f.col('risk_allele').rlike(r'^[ACTG]+$'),
+                f.reverse(f.translate(f.col('risk_allele'), "ACTG", "TGAC"))
+            )
+            .otherwise(f.col('risk_allele'))
+        )
+
+        # Adding columns flagging concordance:
+        .withColumn(
+            'is_concordant',
+            # If risk allele is found on the positive strand:
+            f.when((f.col('risk_allele') == f.col('ref')) | (f.col('risk_allele') == f.col('alt')), True)
+            # If risk allele is found on the negative strand:
+            .when(
+                (f.col('risk_allele_reverse_complement') == f.col('ref'))
+                | (f.col('risk_allele_reverse_complement') == f.col('alt')),
+                True
+            )
+            # If risk allele is ambiguous, still accepted:
+            .when(f.col('risk_allele') == '?', True)
```

Comment:
I don't understand this part. Why we keep the ambiguous alleles?

Reply:
It was in the original logic. I kind of agree with. Question marks are due to how the GWAS Catalog curates risk alleles. They represents ~22% of all associations. For the details you can ask @buniello . 

```python
df = pd.read_csv('https://www.ebi.ac.uk/gwas/api/search/downloads/full', sep='\t')
len(df.loc[df['STRONGEST SNP-RISK ALLELE'].str.endswith('?')])/len(df)
```